### DISCUSSION TITLE
[Question]A Custom Code Theme In Markdown.

### DISCUSSION THREAD

**Discussion by tanmay-bhatgare** [flet-dev/flet#4287] [Q&A] [answered]
### Question

Is there any way to add custom theme for code block in `ft.Markdown()`? I look into `markdown.py` file but didn't find any solution. Your answers and welcomed?

### Code sample

```python
None
```

### Error message

```bash
None
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ingdesarr1** [comment]
In the documentation, appears this:

https://flet.dev/docs/reference/types/markdownstylesheet/

**Reply by tanmay-bhatgare** [reply]
> In the documentation, appears this:
> 
> https://flet.dev/docs/reference/types/markdownstylesheet/

@ingdesarr1 
thanks for help but `ft.MarkdownStyleSheet` is used to style entire markdown control including its style(like h1 text size etc), but I specifically want to add my own custom code theme so idt it's what I need.

**Comment by ndonkoHenri** [comment]
use [`Markdown.code_style_sheet`](https://flet.dev/docs/controls/markdown#code_style_sheet):
```py
def main(page: ft.Page):
    page.scroll = "auto"

    page.add(
        ft.Markdown(
            YOUR_TEXT_HERE,
            selectable=True,
            extension_set=ft.MarkdownExtensionSet.GITHUB_WEB,
            code_theme=ft.MarkdownCodeTheme.ATOM_ONE_DARK,
            code_style_sheet=ft.MarkdownStyleSheet(
                code_text_style=ft.TextStyle(color=ft.colors.YELLOW, size=12),
            ),
            on_tap_link=lambda e: page.launch_url(e.data),
        )
    )


ft.app(main)

```

**Reply by tanmay-bhatgare** [reply]
@ndonkoHenri
Apologies for the late update, but when I tried implementing the syntax highlighting, it just colors every word purple, leaving only strings intact. See:
![Screenshot 2024-11-08 175925](https://github.com/user-attachments/assets/45474529-9196-446c-acf4-fe0310726ed1)
 I’d like to create a custom theme with different colors for keywords, variables, strings, etc., something structured like a dictionary holding these elements. How did you set up themes like the Atom or GitHub code themes initially, and can I do the same and how?

**Reply by ndonkoHenri** [reply, marked as answer]
I created an issue to fix that: https://github.com/flet-dev/flet/issues/4342
